### PR TITLE
Extend Quick Inspector decorations

### PR DESCRIPTION
**Pull request by pasnox**
Now has support for Qt Quick Controls 2 decorations:
- Padding
- background rect (information is known but not used currently)
- contentItem rect (information is known but not used currently)

A legend tool window is also added so we can have a better understanding of what decorations means.
The tool legend window is triggerable from the preview widget toolbar.

The margins and padding decorations no longer have "Margin" or "Padding" labels as they were making the decorations difficult to read.

NOTE: Even though this works was done for QQC2, any item matching their property names (like padding, contentItem etc) would have support for free.

This implement KDEND-133, KDEND-135 and KDEND-136

### CODE REVIEW

File: plugins/quickinspector/quickoverlaylegend.cpp [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+        m_items << Item(
+                       QBrush(),
+                       QuickOverlay::CoordinatesColor,
+                       QT_TR_NOOP("Coordinates (x, y...)")
+                       );
+
+        m_items << Item(
+                       QBrush(),
+                       QuickOverlay::MarginsColor,
+                       QT_TR_NOOP("Margins")
```
Comment (jpetersen) [inline plugins/quickinspector/quickoverlaylegend.cpp]: Maybe better "Margins/Anchors"?

File: plugins/quickinspector/quickoverlay.cpp [hunk 2]
```diff
@@ -124,8 +139,8 @@ bool ItemOrLayoutFacade::isLayout() const
 QuickOverlay::QuickOverlay()
     : m_window(nullptr)
     , m_currentToplevelItem(nullptr)
-    , m_isGrabbingMode(false)
-    , m_drawDecorations(true)
+  , m_isGrabbingMode(false)
```
Comment (jpetersen) [inline plugins/quickinspector/quickoverlay.cpp]: ws?

File: plugins/quickinspector/quickoverlay.cpp [hunk 3]
```diff
@@ -280,7 +295,7 @@ void QuickOverlay::updateOverlay()
     if (m_window) {
         if (!m_currentItem.isNull()) {
             Q_ASSERT(m_currentItem.item()->window() == m_window);
-        }
+}
```
Comment (jpetersen) [inline plugins/quickinspector/quickoverlay.cpp]: ws?

File: plugins/quickinspector/quickoverlay.cpp [hunk 4]
```diff
@@ -296,11 +311,11 @@ void QuickOverlay::itemParentChanged(QQuickItem *parent)
 void QuickOverlay::itemWindowChanged(QQuickWindow *window)
 {
     if (m_window == window) {
-        if (!m_currentItem.isNull())
-            placeOn(m_currentItem);
+    if (!m_currentItem.isNull())
```
Comment (jpetersen) [inline plugins/quickinspector/quickoverlay.cpp]: ws?

File: plugins/quickinspector/quickitemgeometry.cpp [hunk 5]
```diff
@@ -80,6 +86,22 @@ void QuickItemGeometry::initFrom(QQuickItem *item)
     }
     x = item->x();
     y = item->y();
+
+    const QMetaObject *mo = item->metaObject();
+    if (mo->property(mo->indexOfProperty("padding")).isValid()) {
```
Comment (jpetersen) [inline plugins/quickinspector/quickitemgeometry.cpp]: Hm, we only test for "padding" not the other properties

File: plugins/quickinspector/quickitemgeometry.cpp [hunk 6]
```diff
@@ -107,11 +112,12 @@ void QuickItemGeometry::scaleTo(qreal factor)
     baselineOffset = baselineOffset * factor;
     x = x * factor;
     y = y * factor;
-}
+    }
```
Comment (jpetersen) [inline plugins/quickinspector/quickitemgeometry.cpp]: ws?

**Comment by jpetersen** [comment]
Just the little things, else looks good

File: plugins/quickinspector/quickoverlaylegend.cpp (hunk 1 above)
Comment (pasnox) [inline plugins/quickinspector/quickoverlaylegend.cpp]: Done

File: plugins/quickinspector/quickoverlay.cpp (hunk 2 above)
Comment (pasnox) [inline plugins/quickinspector/quickoverlay.cpp]: Fixed in latest commit

File: plugins/quickinspector/quickoverlay.cpp (hunk 3 above)
Comment (pasnox) [inline plugins/quickinspector/quickoverlay.cpp]: Done

File: plugins/quickinspector/quickoverlay.cpp (hunk 4 above)
Comment (pasnox) [inline plugins/quickinspector/quickoverlay.cpp]: Done

File: plugins/quickinspector/quickitemgeometry.cpp (hunk 5 above)
Comment (pasnox) [inline plugins/quickinspector/quickitemgeometry.cpp]: Others like background and contentItem are checked from line 57-62.
Do I missed some other properties ?

File: plugins/quickinspector/quickitemgeometry.cpp (hunk 6 above)
Comment (pasnox) [inline plugins/quickinspector/quickitemgeometry.cpp]: Done

**Pull request merged**